airsenal/framework/utils.py

```diff
@@ -238,19 +238,27 @@ def get_sell_price_for_player(player_id, gameweek=None, fpl_team_id=None):
     if the price increased in that time, we only get half the profit.
     if gameweek is None, get price we could sell the player for now.
     """
-    if not fpl_team_id:
-        fpl_team_id = fetcher.FPL_TEAM_ID
+    if fpl_team_id:
+        fetcher = FPLDataFetcher(fpl_team_id) # create new fetcher specific to team id
+    else:
+        fpl_team_id = fetcher.FPL_TEAM_ID #use saved fetcher with default team id.
+    
     transactions = session.query(Transaction)
     transactions = transactions.filter_by(fpl_team_id=fpl_team_id)
     transactions = transactions.filter_by(player_id=player_id)
     transactions = transactions.order_by(Transaction.gameweek).all()
 
+    history =fetcher.get_fpl_team_history_data()
+    starting_gw_adj = history['current'][0]['event'] - 1 #if you make a squad after then first gameweek then in transactions 'gameweek 1' is your starting gameweek
```

Comment:
This is probably something we should fix when filling the database with initial squads, so the gameweeks in the transactions table are always the actual gameweeks and it doesn't need to be fixed here.